ambry-frontend/src/test/java/com/github/ambry/frontend/NamedBlobPutHandlerTest.java

```diff
@@ -192,6 +193,26 @@ public void updateNamedBlobAllowedTest() throws Exception {
     assertEquals("Unexpected response status", restResponseChannel.getStatus(), ResponseStatus.Ok);
   }
 
+  @Test
+  public void testStripPrefixAndExtension() throws Exception {
+    String EXTENSION = ".bin";
+    Properties properties = new Properties();
+    properties.setProperty("frontend.path.prefixes.to.remove", PATH_PREFIX_TO_REMOVE);
+    initNamedBlobPutHandler(properties);
+
+    String blobId = CLUSTER_NAME + CONVERTED_ID + EXTENSION;
+    assertEquals("Blob Id should not have prefix and extention", RestUtils.stripSlashAndExtensionFromId(CONVERTED_ID),
+        stripPrefixAndExtension(blobId));
+
+    blobId = PATH_PREFIX_TO_REMOVE + "/" + CLUSTER_NAME + CONVERTED_ID + EXTENSION;
+    assertEquals("Blob Id should not have prefix and extention", RestUtils.stripSlashAndExtensionFromId(CONVERTED_ID),
+        stripPrefixAndExtension(blobId));
+
+    blobId = CONVERTED_ID + EXTENSION;
+    assertEquals("Blob Id should not have prefix and extention", RestUtils.stripSlashAndExtensionFromId(CONVERTED_ID),
```

Comment:
why are we comparing it the RestUtils.stripSlashAndExtensionFromId, shouldn't we compare it with CONVERTED_ID?

Reply:
hmmm. it's because the CONVERTED_ID has the leading slash, so I didn't add another "/" into the request path. Updated the test a little bit to avoid confusion.